go/cmd/vtexplain/vtexplain.go

```diff
@@ -29,63 +28,51 @@ import (
+	querypb "vitess.io/vitess/go/vt/proto/query"
 )
 
 var (
-	sqlFlag            = flag.String("sql", "", "A list of semicolon-delimited SQL commands to analyze")
-	sqlFileFlag        = flag.String("sql-file", "", "Identifies the file that contains the SQL commands to analyze")
-	schemaFlag         = flag.String("schema", "", "The SQL table schema")
-	schemaFileFlag     = flag.String("schema-file", "", "Identifies the file that contains the SQL table schema")
-	vschemaFlag        = flag.String("vschema", "", "Identifies the VTGate routing schema")
-	vschemaFileFlag    = flag.String("vschema-file", "", "Identifies the VTGate routing schema file")
-	ksShardMapFlag     = flag.String("ks-shard-map", "", "JSON map of keyspace name -> shard name -> ShardReference object. The inner map is the same as the output of FindAllShardsInKeyspace")
-	ksShardMapFileFlag = flag.String("ks-shard-map-file", "", "File containing json blob of keyspace name -> shard name -> ShardReference object")
-	numShards          = flag.Int("shards", 2, "Number of shards per keyspace. Passing --ks-shard-map/--ks-shard-map-file causes this flag to be ignored.")
-	executionMode      = flag.String("execution-mode", "multi", "The execution mode to simulate -- must be set to multi, legacy-autocommit, or twopc")
-	replicationMode    = flag.String("replication-mode", "ROW", "The replication mode to simulate -- must be set to either ROW or STATEMENT")
-	normalize          = flag.Bool("normalize", false, "Whether to enable vtgate normalization")
-	outputMode         = flag.String("output-mode", "text", "Output in human-friendly text or json")
-	dbName             = flag.String("dbname", "", "Optional database target to override normal routing")
-	plannerVersionStr  = flag.String("planner-version", "", "Sets the query planner version to use when generating the explain output. Valid values are V3 and Gen4")
-
-	// vtexplainFlags lists all the flags that should show in usage
-	vtexplainFlags = []string{
-		"output-mode",
-		"planner-version",
-		"normalize",
-		"shards",
-		"replication-mode",
-		"schema",
-		"schema-file",
-		"sql",
-		"sql-file",
-		"vschema",
-		"vschema-file",
-		"ks-shard-map",
-		"ks-shard-map-file",
-		"dbname",
-		"queryserver-config-passthrough-dmls",
-	}
+	sqlFlag            string
+	sqlFileFlag        string
+	schemaFlag         string
+	schemaFileFlag     string
+	vschemaFlag        string
+	vschemaFileFlag    string
+	ksShardMapFlag     string
+	ksShardMapFileFlag string
+	replicationMode    string
+	normalize          bool
+	dbName             string
+	plannerVersionStr  string
+
+	numShards     = 2
+	executionMode = "multi"
+	outputMode    = "text"
 )
 
+func registerFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&sqlFlag, "sql", "", "A list of semicolon-delimited SQL commands to analyze")
+	fs.StringVar(&sqlFileFlag, "sql-file", "", "Identifies the file that contains the SQL commands to analyze")
+	fs.StringVar(&schemaFlag, "schema", "", "The SQL table schema")
```

Comment:
these should still use the variable as the "default value" argument, for example:

```diff
- fs.StringVar(&sqlFlag, "sql", "", "A list of semicolon-delimited SQL commands to analyze")
+ fs.StringVar(&sqlFlag, "sql", sqlFlag, "A list of semicolon-delimited SQL commands to analyze")
```

the reason being if we ever change the variable default, then we don't need to remember to also update the flag default, since they are inextricably linked

Reply:
👍